pennylane/ops/functions/equal.py

```diff
@@ -223,6 +223,41 @@ def _equal_controlled(op1: Controlled, op2: Controlled, **kwargs):
         ) from e
 
 
+@_equal.register
+# pylint: disable=unused-argument
+def _equal_pow(op1: Pow, op2: Pow, **kwargs):
```

Comment:
As we chatted, maybe adding a note on the fact that the attributes of the operators are being checked could be useful just to help users.